app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,54 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.all.find_by(id: task_id)
+
+    if @task.nil?
+      head :not_found
+    end
+
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(action: params[:task][:action], completed_date: params[:task][:completed_date])
```

Comment:
This is pretty good, but it could be shorter and safer with using the `task_params` method you defined below. Just like how you used `task_params` in `update` on line 36, you could change the line above to
```ruby
@task = Task.new(task_params)
```